Core/include/Acts/Vertexing/IVertexFinder.hpp

```diff
@@ -0,0 +1,40 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2023 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Utilities/Any.hpp"
+#include "Acts/Utilities/Result.hpp"
+#include "Acts/Vertexing/VertexingOptions.hpp"
+
+#include <vector>
+
+namespace Acts {
+
+class Vertex;
+struct InputTrack;
+struct VertexingOptions;
+
+class IVertexFinder {
+ public:
+  using State = Acts::AnyBase<128>;
+
+  virtual Result<std::vector<Vertex>> find(
+      const std::vector<InputTrack>& trackVector,
+      const VertexingOptions& vertexingOptions, State& state) const = 0;
+
+  virtual State makeState() const = 0;
```

Comment:
I guess this is like a default construction right? So any initialization depending on inputs has to be done afterwards?

Reply:
Yes and no. Construction can depend on finder config arguments that are stored in the finder instance.

That's sort of what `hasTrivialState` is supposed to communicate, but maybe too complex. 

The basic idea is that it doesn't have to have any external input, and the state can be created by the finder implementation.